api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/DQACargaViralCohortQueries.java

```diff
@@ -85,4 +116,145 @@ public DataDefinition getAge(String calculateAgeOn) {
     spdd.setQuery(substitutor.replace(sql));
     return spdd;
   }
+
+  public DataDefinition getDataNotificouCV() {
+
+    SqlPatientDataDefinition spdd = new SqlPatientDataDefinition();
+
+    spdd.setName("Data de Consulta onde Notificou o Resultado de CV");
+    spdd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    spdd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    spdd.addParameter(new Parameter("location", "Location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+
+    map.put("856", hivMetadata.getHivViralLoadConcept().getConceptId());
+    map.put("1305", hivMetadata.getHivViralLoadQualitative().getConceptId());
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+
+    String query =
+        "SELECT patient_id, MIN(first_result_date) AS lab_result_date "
+            + "            FROM( "
+            + "               SELECT p.patient_id, Min(e.encounter_datetime) AS first_result_date "
+            + "               FROM   patient p "
+            + "                   INNER JOIN encounter e ON p.patient_id = e.patient_id "
+            + "                   INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "               WHERE p.voided = 0 "
+            + "                 AND e.voided = 0 "
+            + "                 AND o.voided = 0 "
+            + "                 AND e.location_id = :location "
+            + "                 AND e.encounter_type = ${6} "
+            + "                 AND o.concept_id = ${856} "
+            + "                 AND o.value_numeric IS NOT NULL "
+            + "                 AND e.encounter_datetime >= :startDate "
+            + "                 AND e.encounter_datetime <= DATE_SUB(DATE_ADD(:startDate, INTERVAL 1 MONTH), INTERVAL 1 DAY) "
+            + "               GROUP  BY p.patient_id "
+            + "            UNION "
+            + "            SELECT p.patient_id, Min(e.encounter_datetime) AS first_result_date "
+            + "            FROM   patient p "
+            + "                   INNER JOIN encounter e ON p.patient_id = e.patient_id "
+            + "                   INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "            WHERE  p.voided = 0 "
+            + "                   AND e.voided = 0 "
+            + "                   AND o.voided = 0 "
+            + "                   AND e.location_id = :location "
+            + "                   AND e.encounter_type = ${6} "
+            + "                   AND o.concept_id = ${856} "
+            + "                   AND o.value_numeric IS NOT NULL "
+            + "                   AND e.encounter_datetime >= DATE_ADD(:startDate, INTERVAL 1 MONTH) "
```

Comment:
Instead of joining tables with UNION clause, you can remove the UNION and the repeated part of query and use the OR clause to join the date ranges.
eg: select ... where... and ((First range) OR (Second range) OR (third range))